rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/ExportToCatalogConfirmWindow.java

```diff
@@ -0,0 +1,356 @@
+        this.commitMessage = new TextBox();
+        exportInfoPanel.add(this.commitMessage);
+
+        this.revisionLabel = new Label(INITIAL_COMMIT_MESSAGE);
+        this.revisionLabel.setHeight("80px");
+        this.revisionLabel.setValign(VerticalAlignment.TOP);
+        this.revisionLabel.setAlign(Alignment.LEFT);
+        exportInfoPanel.add(this.revisionLabel);
+
+        HLayout buttons = new HLayout();
+        buttons.setHeight(40);
+        buttons.setMembersMargin(5);
+        buttons.setAlign(Alignment.RIGHT);
+        buttons.setAlign(VerticalAlignment.BOTTOM);
+
+        IButton ok = new IButton("OK", event -> submitNodeSourceConfiguration());
+        ok.setIcon(Images.instance.ok_16().getSafeUri().asString());
+        IButton cancel = new IButton("Cancel", event -> hideAndDestroy(this));
+        cancel.setIcon(Images.instance.cancel_16().getSafeUri().asString());
+        buttons.setMembers(ok, cancel);
+
+        VLayout layout = new VLayout();
+        layout.setAlign(VerticalAlignment.TOP);
+        layout.setMargin(10);
+        layout.addMember(this.windowLabel);
+        layout.addMember(exportInfoPanel);
+        layout.addMember(buttons);
+
+        addItem(layout);
+    }
+
+    private void fillBucketList() {
+        this.bucketList.setEnabled(false);
+        this.bucketList.addItem(SELECT_A_BUCKET_OPTION);
+        this.bucketList.addChangeHandler(event -> requestNodeSourceInBucket(this.bucketList.getSelectedValue()));
+        RequestBuilder request = new RequestBuilder(RequestBuilder.GET,
+                                                    new CatalogUrlBuilder().getCatalogUrl() + "/buckets");
+        request.setHeader("sessionId", LoginModel.getInstance().getSessionId());
+        request.setCallback(fillBucketListWithRequestCallback());
+        try {
+            request.send();
+        } catch (RequestException e) {
+            String errorMessage = "Request sent to catalog failed";
+            GWT.log(errorMessage, e);
+            this.windowLabel.setContents(errorMessage);
+        }
+    }
+
+    private void requestNodeSourceInBucket(String bucketName) {
+        if (!bucketName.equals(SELECT_A_BUCKET_OPTION)) {
+            CatalogRequestBuilder catalogRequestBuilder = new CatalogRequestBuilder(null);
+            catalogRequestBuilder.requestNodeSourcesForBucket(bucketName, new RequestCallback() {
+                @Override
+                public void onResponseReceived(Request request, Response response) {
+                    JSONArray nodeSources = JSONParser.parseStrict(response.getText()).isArray();
+                    isRevision = false;
+                    revisionLabel.setContents("(Initial commit)");
+                    for (int i = 0; i < nodeSources.size(); i++) {
+                        JSONObject nodeSource = nodeSources.get(i).isObject();
+                        String nodeSourceName = nodeSource.get("name").isString().stringValue();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeSourceName" which hides the field declared at line 78. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)